app/bundles/PageBundle/EventListener/BuilderSubscriber.php

```diff
@@ -108,9 +115,21 @@ public function onPageBuild(Events\PageBuilderEvent $event)
             $event->addAbTestWinnerCriteria('page.dwelltime', $dwellTime);
         }
 
-        if ($event->tokensRequested([$this->pageTokenRegex])) {
+        if ($event->tokensRequested([$this->pageTokenRegex, $this->dwcTokenRegex])) {
             $event->addTokensFromHelper($tokenHelper, $this->pageTokenRegex, 'title', 'id', false, true);
 
+            // add only filter based dwc tokens
+            $dwcTokenHelper = new BuilderTokenHelper($this->factory, 'dynamicContent');
```

Comment:
Why the new code uses `factory`? It should be deprecated?

Reply:
It is; it's because BuilderTokenHelper has been around for a while and still depends on it. We haven't refactored it to be a service.